browser/ui/views/split_view/split_view_location_bar.cc

```diff
@@ -0,0 +1,263 @@
+  cc::PaintFlags flags;
+  flags.setColor(cp->GetColor(kColorBraveSplitViewInactiveWebViewBorder));
+  flags.setAntiAlias(true);
+  flags.setStyle(cc::PaintFlags::kStroke_Style);
+  flags.setStrokeWidth(2);
+  canvas->DrawPath(path, flags);
+}
+
+gfx::Size SplitViewLocationBar::CalculatePreferredSize(
+    const views::SizeBounds& available_size) const {
+  auto size = views::WidgetDelegateView::CalculatePreferredSize(available_size);
+  if (auto* view = view_observation_.GetSource(); view && view->width()) {
+    size.set_width(std::min(view->width(), size.width()));
+  }
+  return size;
+}
+
+void SplitViewLocationBar::PrimaryPageChanged(content::Page& page) {
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::DidChangeVisibleSecurityState() {
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::WebContentsDestroyed() {
+  Observe(nullptr);
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::OnViewVisibilityChanged(views::View* observed_view,
+                                                   views::View* starting_view) {
+  UpdateVisibility();
+}
+
+void SplitViewLocationBar::OnViewBoundsChanged(views::View* observed_view) {
+  UpdateBounds();
+}
+
+void SplitViewLocationBar::OnViewIsDeleting(views::View* observed_view) {
+  view_observation_.Reset();
+}
+
+void SplitViewLocationBar::UpdateVisibility() {
+  auto* view = view_observation_.GetSource();
+  if (view && view->GetVisible()) {
+    GetWidget()->ShowInactive();
+  } else {
+    GetWidget()->Hide();
+  }
+}
+
+void SplitViewLocationBar::UpdateBounds() {
+  auto* view = view_observation_.GetSource();
+  if (!view) {
+    return;
+  }
+
+  gfx::Point point;
+  views::View::ConvertPointToWidget(view, &point);
+
+  auto* widget = GetWidget();
+  CHECK(widget);
+  GetWidget()->SetBounds(gfx::Rect(point, GetPreferredSize()));
+}
+
+void SplitViewLocationBar::UpdateURLAndIcon() {
+  if (auto* contents = web_contents()) {
+    auto url = contents->GetVisibleURL();
+    auto origin = url::Origin::Create(url);
+    auto formatted_origin = url_formatter::FormatUrl(
+        origin.opaque() ? url : origin.GetURL(),
+        url_formatter::kFormatUrlOmitDefaults |
+            url_formatter::kFormatUrlOmitHTTPS,
+        base::UnescapeRule::SPACES, nullptr, nullptr, nullptr);
+    url_->SetText(formatted_origin);
+    UpdateIcon();
+  } else {
+    url_->SetText({});
+    UpdateIcon();
+  }
+
+  UpdateBounds();
+}
+
+void SplitViewLocationBar::UpdateIcon() {
+  // At the moment, we show only warning icon.
+  safety_icon_->SetVisible(!IsContentsSafe());
+}
+
+bool SplitViewLocationBar::IsContentsSafe() const {
+  if (!web_contents()) {
+    return true;
+  }
+
+  if (SecurityStateTabHelper::FromWebContents(web_contents())
+          ->GetSecurityLevel() == security_state::SecurityLevel::SECURE) {
+    return true;
+  }
+
+  auto url = web_contents()->GetVisibleURL();
+  return url.is_empty() || url.SchemeIs(url::kAboutScheme) ||
+         url.SchemeIs("chrome");
```

Comment:
we probably have to include a bunch more protocols here, like blob, data, file, etc. not sure if there's a better way to do it because it's hard to enumerate all of them.

i guess though if we are showing the full origin, including scheme, as i suggested above, there's no need to show any security indicator unless it's an invalid HTTPS page (in which case we can put a slash through the HTTPS as is done in the url bar)

Reply:
Just as a quick sanity check - is it easier to check insecure schemes? (http, https ect)?

Reply:
Yeah, The latest version is checking like

```
  auto security_level = SecurityStateTabHelper::FromWebContents(web_contents())
                            ->GetSecurityLevel();
  return security_level == security_state::SecurityLevel::SECURE ||
         security_level == security_state::SecurityLevel::NONE;
```

The NONE type would include them, so we can filter only the insecure state.